components/TextFields/src/MDCTextInputControllerLegacyDefault.m

```diff
@@ -0,0 +1,1261 @@
+}
+
+static inline UIColor *MDCTextInputDefaultActiveColorDefault() {
+  return [MDCPalette bluePalette].accent700;
+}
+
+static inline UIColor *MDCTextInputDefaultNormalUnderlineColorDefault() {
+  return [UIColor lightGrayColor];
+}
+
+static inline UIColor *MDCTextInputDefaultTextErrorColorDefault() {
+  return [MDCPalette redPalette].accent400;
+}
+
+#pragma mark - Class Properties
+
+static BOOL _floatingEnabledDefault = YES;
+static BOOL _mdc_adjustsFontForContentSizeCategoryDefault = YES;
+
+static CGFloat _floatingPlaceholderScaleDefault =
+    MDCTextInputDefaultFloatingPlaceholderScaleDefault;
+
+static UIColor *_activeColorDefault;
+static UIColor *_disabledColorDefault;
+static UIColor *_errorColorDefault;
+static UIColor *_floatingPlaceholderColorDefault;
+static UIColor *_inlinePlaceholderColorDefault;
+static UIColor *_normalColorDefault;
+
+static UITextFieldViewMode _underlineViewModeDefault = UITextFieldViewModeWhileEditing;
+
+@interface MDCTextInputControllerLegacyDefault () {
+  BOOL _mdc_adjustsFontForContentSizeCategory;
+
+  UIColor *_activeColor;
+  UIColor *_disabledColor;
+  UIColor *_floatingPlaceholderColor;
+  UIColor *_inlinePlaceholderColor;
+  UIColor *_normalColor;
+}
+
+@property(nonatomic, assign, readonly) BOOL isDisplayingCharacterCountError;
+@property(nonatomic, assign, readonly) BOOL isDisplayingErrorText;
+@property(nonatomic, assign, readonly) BOOL isPlaceholderUp;
+@property(nonatomic, assign) BOOL isRegisteredForKVO;
+
+@property(nonatomic, strong) MDCTextInputAllCharactersCounter *internalCharacterCounter;
+
+@property(nonatomic, strong) NSArray<NSLayoutConstraint *> *placeholderAnimationConstraints;
+
+@property(nonatomic, strong) NSLayoutConstraint *characterCountY;
+@property(nonatomic, strong) NSLayoutConstraint *characterCountTrailing;
+@property(nonatomic, strong) NSLayoutConstraint *clearButtonY;
+@property(nonatomic, strong) NSLayoutConstraint *clearButtonTrailingCharacterCountLeading;
+@property(nonatomic, strong) NSLayoutConstraint *placeholderLeading;
+@property(nonatomic, strong) NSLayoutConstraint *placeholderTop;
+@property(nonatomic, strong) NSLayoutConstraint *placeholderTrailingCharacterCountLeading;
+@property(nonatomic, strong) NSLayoutConstraint *placeholderTrailingSuperviewTrailing;
+
+@property(nonatomic, copy) NSString *errorAccessibilityValue;
+@property(nonatomic, copy, readwrite) NSString *errorText;
+@property(nonatomic, copy) NSString *previousLeadingText;
+
+@property(nonatomic, strong) UIColor *previousPlaceholderColor;
+
+@property(nonatomic, strong) UIFont *customLeadingFont;
+@property(nonatomic, strong) UIFont *customPlaceholderFont;
+@property(nonatomic, strong) UIFont *customTrailingFont;
+
+@end
+
+@implementation MDCTextInputControllerLegacyDefault
+
+@synthesize characterCounter = _characterCounter;
+@synthesize characterCountMax = _characterCountMax;
+@synthesize characterCountViewMode = _characterCountViewMode;
+@synthesize errorColor = _errorColor;
```

Comment:
If you explicitly define setErrorColor and errorColor below do you need to @synthesize?
(Here and above / below)